### PR TITLE
Add code to be served via WASM and the relevant automation

### LINKED ISSUE #1: Go: Add code to be served via WASM and the relevant automation
- [x] <b>Description</b>
    We want to be able to serve Go code to the browser via WASM. <br>
- [x] <b>Use cases</b>
    1. Use Go code to process data in the browser <br>
    2. No need to maintain separate front end code.
- [x] <b>Proposed design</b>
    https://github.com/golang/go/wiki/WebAssembly
- [ ] <b>Alternative design approaches</b>
    NA.
- [ ] <b>Workaround</b>
    NA.
- [ ] <b>Additional information</b>
    NA.

### PR DESCRIPTION
**Pull request by the-c0d3br34k3r**
## Description
Closes #1.
Add code to be served via WASM and the relevant automation

## Author's Checklist
- [x] **These changes don't break existing dependants.** If they do, and you suggest to proceed, 
please explain why.

## Reviewer's Checklist
- [ ] **All classes have documentation comments.** - NA
- [x] **All methods have documentation comments.**
- [x] **No parameters are hardcoded.** Should be loaded as properties instead. If there needs to be hardcoded properties, explain why.
- [ ] **All information to be logged/displayed to the user are internationalized.** If not, explain why.  - NA
- [x] **README.md has been updated if needed.**
- [ ] **pom.xml version is set to the latest date** If not, explain why. - NA

## Assignee's Checklist
- [ ] **All GitHub action checks have passed.** - Snyk Failures are for indirect dependencies that we don't have control over
- [x] **All reviewers have approved.**
- [x] **Relevant documentation has been updated if needed.**
- [ ] **pom.xml version is set to the latest date** If not, explain why. - NA

### CODE REVIEW

File: add_test.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package gopopper
+package main
 
 import (
```
Comment (abprash) [inline add_test.go:3]: should this file be removed, considering `add.go` is deleted ?

File: add_wasm.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	number1Input := document.Call("getElementById", "number1")
+	number2Input := document.Call("getElementById", "number2")
+	fmt.Println("number 2:", number2Input.Get("value"))
+	var callBack js.Func = js.FuncOf(func(this js.Value, args []js.Value) any {
+		AddAndSetResult()
+		return nil
+	})
+	number1Input.Call("addEventListener", "change", callBack)
+	number2Input.Call("addEventListener", "change", callBack)
+	select {}
```
Comment (abprash) [inline add_wasm.go:61]: what does this do ?

File: add_wasm.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+// Gets a numeric value from an HTML element body given an ID.
+func getNumberFromElement(id string) (int, error) {
+	element := getElementById(id)
+	valueStr := element.Get("value").String()
+	return strconv.Atoi(valueStr)
+}
+
+// Adds the two input elements and sets the sum to another element.
+func AddAndSetResult() {
```
Comment (abprash) [inline add_wasm.go]: minor: why is `Add`, `AddAndSetResult` capitalized while others aren't ?

File: add_wasm.go [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	number2, err := getNumberFromElement("number2")
+	if err != nil {
+		sumNode.Set("innerHTML", err.Error())
+		fmt.Println(err)
+		return
+	}
+	sumNode.Set("innerHTML", Add(number1, number2))
+}
+
+// Start of execution.
```
Comment (abprash) [inline add_wasm.go:44]: minor: do these comments add value ?

File: add_wasm.go (hunk 2 above)
Comment (the-c0d3br34k3r) [reply add_wasm.go:61]: This is like an infinite for loop. It prevents the Go code from exiting. We need this as the code deals with async events.

File: add_wasm.go (hunk 4 above)
Comment (the-c0d3br34k3r) [reply add_wasm.go:44]: It's just the Go template for function doc comments. Keeping it in lieu of format. No functional value in this case.

File: add_wasm.go (hunk 3 above)
Comment (the-c0d3br34k3r) [reply add_wasm.go]: Per convention, internal methods that won't be imported start with a lower case (`addAndSetResult` etc).
But methods that will be imported outside the file (`Add`) will start from upper case.

Have changed `AddAndSetResult ` -> `addAndSetResult `

Comment (abprash) [reply add_wasm.go]: Good to know. 👍 
Thanks for updating.

**Review by abprash — APPROVED** [review]

**Pull request merged**